lib/srv/forward/sshserver.go

```diff
@@ -851,6 +850,84 @@ func (s *Server) handleConnection(ctx context.Context, chans <-chan ssh.NewChann
 	}
 }
 
+// handleClientChannels handles channel open requests from the remote server.
+func (s *Server) handleClientChannels(ctx context.Context) {
+	forwardedTCPIP := s.remoteClient.HandleChannelOpen(teleport.ChanForwardedTCPIP)
+
+	for nch := range forwardedTCPIP {
+		if nch == nil {
+			continue
+		}
+		chanCtx, nch := tracessh.ContextFromNewChannel(nch)
+		ctx, span := s.tracerProvider.Tracer("ssh").Start(
+			oteltrace.ContextWithRemoteSpanContext(ctx, oteltrace.SpanContextFromContext(chanCtx)),
+			fmt.Sprintf("ssh.Forward.OpenChannel/%s", nch.ChannelType()),
+			oteltrace.WithSpanKind(oteltrace.SpanKindServer),
+			oteltrace.WithAttributes(
+				semconv.RPCServiceKey.String("ssh.ForwardServer"),
+				semconv.RPCMethodKey.String("OpenChannel"),
+				semconv.RPCSystemKey.String("ssh"),
+			),
+		)
+
+		go func() {
+			defer span.End()
+			s.handleForwardedTCPIPRequest(ctx, nch)
+		}()
+	}
+}
+
+// handleForwardedTCPIPRequest handles remote port forwarding requests.
+func (s *Server) handleForwardedTCPIPRequest(ctx context.Context, nch ssh.NewChannel) {
```

Comment:
We're not rejecting the channel open in case of early errors; even just a `defer nch.Reject(...)` would do, because `Reject` will just return an error if it's called on a newchannel that was already accepted or rejected.